File: backend/project/src/main/java/com/lguplus/project/order/domain/Order.java
package com.lguplus.project.order.domain;

import com.lguplus.project.device.domain.Device;
import com.lguplus.project.plan.domain.Plan;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long orderNumber;

    @Enumerated(EnumType.STRING)
    private JoinType joinType;

    private LocalDate joinDate;
    private Integer monthlyFee;

    @Enumerated(EnumType.STRING)
    private DiscountType discountType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(referencedColumnName = "code",
                name = "device_code")
    @JoinColumn(referencedColumnName = "name",
            name = "device_name")
    private Device device;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(referencedColumnName = "name",
                name = "plan_name")
    private Plan plan;

    private String color;

    @Column(length = 1000)
    private String picPaths;
}

File: frontend/project/src/api/order.ts
import orderApi from '../lib/customApi'

export async function getOrderList() {
  const response = await orderApi.get<OrderListData>(`/api/order`)
  return response.data
}

export async function saveOrder(orderRequest: OrderRequest) {
  const response = await orderApi.post<OrderData>(`/api/order`, orderRequest)
  return response.data
}

export async function deleteOrder(orderNumber: string) {
  const response = await orderApi.delete(`/api/order/${orderNumber}`)
  return response.data
}

export interface OrderRequest {
  deviceCode: string
  planName: string
  joinType: string
  monthlyFee: number
  discountType: string
  color: string
}

export interface OrderData extends OrderRequest {
  orderNumber: string
  joinDate: string
  deviceName: string
  picPaths: Array<string>
}

export interface OrderListData extends Array<OrderData> {}
